app/v1/controllers/requestParsers/validators/ListPaymentsValidator.scala

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v1.controllers.requestParsers.validators
+
+import v1.controllers.requestParsers.validators.validations._
+import v1.models.errors._
+import v1.models.request.listPayments.ListPaymentsRawRequest
+
+class ListPaymentsValidator extends Validator[ListPaymentsRawRequest] {
+  private val validationSet = List(parameterFormatValidation, parameterRuleValidation)
+
+  private def parameterFormatValidation: ListPaymentsRawRequest => List[List[MtdError]] = (data: ListPaymentsRawRequest) => {
+    List(
+      NinoValidation.validate(data.nino),
+      data.from.map(DateFormatValidation.validate(_, FromDateFormatError)).getOrElse(Nil),
+      data.to.map(DateFormatValidation.validate(_, ToDateFormatError)).getOrElse(Nil)
+    )
+  }
+
+  private def parameterRuleValidation: ListPaymentsRawRequest => List[List[MtdError]] = { data =>
+    List(
+      MissingParameterValidation.validate(data.from, MissingFromDateError),
```

Comment:
Missing Date error may need to check before Format Date error!!!

Reply:
DateFormatValidation won't be run in the first place if the date is `None`.

Reply:
Happy to move this into parameterFormatValidation if you think it'd be better there though.

Reply:
Why do we need to test the format if the date is not provided? I thought it would be nice if we test the missing date before checking the format. That's my opinion... better check with the team and decide

Reply:
We don't test the format if the date is not provided - `data.from` and `data.to` are Options, so if they are missing then there's nothing to map into and `DateFormatValidation.validate` isn't run.

I don't have an opinion on what should be run first though; I only put the format stuff in the format `def` and the missing stuff separately because I think that's what we've done in the past..